app/services/repository_zip_export.rb

```diff
@@ -5,16 +5,25 @@
 module RepositoryZipExport
   def self.generate_zip(params, repository, current_user)
     # Fetch rows in the same order as in the currently viewed datatable
-    ordered_row_ids = params[:row_ids]
-    id_row_map = RepositoryRow.where(id: ordered_row_ids,
-                                     repository: repository)
-                              .index_by(&:id)
-    ordered_rows = ordered_row_ids.collect { |id| id_row_map[id.to_i] }
+    if params[:my_module_id]
```

Comment:
Should we write a test for this part with a context `when receive RepositorySnapshot`?

Reply:
We need to refactor factories, for adding snapshots. I think better to make it in scope of the separate task.